docs/runtimes/load-from-hub.md

```diff
@@ -4,29 +4,84 @@
 
 **In this section**
-- [Overview](#overview)
-- [Function hub](#function-hub)
-- [Searching for functions](#searching-for-functions)
+- [MLRun Function hub](#mlrun-function-hub)
+- [Private function hub](#private-function-hub)
 - [Setting the project configuration](#setting-the-project-configuration)
 - [Loading functions from the hub](#loading-functions-from-the-hub)
 - [View the function params](#view-the-function-params)
 - [Running the function](#running-the-function)
 
-## Function hub
+## MLRun function hub
 
 The MLRun function hub has a wide range of functions that can be used for a variety of use cases.
 There are functions for ETL, data preparation, training (ML & Deep learning), serving, alerts and notifications and more.
 Each function has a docstring that explains how to use it. In addition, the functions are associated with categories to make it easier for you to find the relevant one.
 
 Functions can be easily imported into your project and therefore help you to speed up your development cycle by reusing built-in code.
 
-## Searching for functions
-
-The Function Hub is located [here](https://www.mlrun.org/hub/). <br>
+The function hub is located [here](https://www.mlrun.org/hub/). <br>
 You can search and filter the categories and kinds to find a function that meets your needs.
 
 ![Hub](../_static/images/marketplace-ui.png)
 
+## Private function hub
+You can create your own function hub, and connect it to MLRun. Then you can import functions (with their tags) from your custom hub.
+
+### Create a custom hub
+
+You can either fork the [MLRun function hub repo](https://github.com/mlrun/functions) and add to it your Git repo, or create a hub from scratch.
+Read [CONTRIBUTING.md](https://github.com/mlrun/functions) to learn how to create a function.
+```{Note}
+Make sure your hub source is accessible via github (private is also possible).
+```
+
+To create a function hub from scratch, the hub structure must be the same as the [MLRun hub](https://github.com/mlrun/marketplace).
+
+The hierarchy must be:
+
+- functions directory
+	- channels directories
+		- some-function-1
+		- some-function-2
+		- ...
+		- some-function-n
+			- version-1
+			- ...
+			- version-n
+			- latest
+				- src
+					- function.yaml
+					- item.yaml
+					- function.py
+					- ...
+				- static (optional)
+					- html files
+					
+
+
+
+### Add a custom hub to the MLRun database
+To add your hub to your MLRun database, run:
+```python
+import mlrun.common.schemas
+
+# Add a private source as the last one (will be #1 in the list)
+private_source = mlrun.common.schemas.IndexedHubSource(
+    order=-1,
+	source=mlrun.common.schemas.HubSource(
+		metadata=mlrun.common.schemas.HubObjectMetadata(
+		name="priv", description="a private source"
+		),
+		spec=mlrun.common.schemas.HubSourceSpec(
+		path="/local/path/to/source", channel="development"
+		),
+	)
```

Comment:
I don't really follow the example and if I struggle, I believe it needs some clarifications.
1. `order=-1` means that it replaces the default IIUC. It should be specified (@yonishelach please confirm)
2. what is local path to source? I expected this to be a url to a git repo. Does local path means its available only for the current user? why do you need a channel if the path is local? @yonishelach
3. please fix the source name to make it a proper example (is priv short for private?)

Reply:
I took it from the documentation of `create_hub_source` so maybe it is need to be improved as well or I am missing something too. According the documentation there:
```
The global hub source always exists in the list, and is always the last source
(`order = -1`). It cannot be modified nor can it be moved to another order in the list.
```
So is it impossible to assign order=-1 at all?
@theSaarco can you please help us here?

```suggestion
# Add a private source as the last one (will be #1 in the list) TODO: need to be resolved
private_source = mlrun.common.schemas.IndexedHubSource(
    order=-1,
	source=mlrun.common.schemas.HubSource(
		metadata=mlrun.common.schemas.HubObjectMetadata(
		name="private", description="a private source"
		),
		spec=mlrun.common.schemas.HubSourceSpec(
		path="https://mlrun.github.io/marketplace", channel="development"
		),
	)
```

Reply:
The problem is that you're quoting just part of the docstring. The `create_hub_source` docstring says:
```
MLRun maintains an ordered list of hub sources (“sources”) Each source has
its details registered and its order within the list. When creating a new source, the special order ``-1``
can be used to mark this source as last in the list. However, once the source is in the MLRun list,
its order will always be ``>0``.

The global hub source always exists in the list, and is always the last source
(``order = -1``). It cannot be modified nor can it be moved to another order in the list.
```

I think this should answer your questions. The `-1` is just for the creation flow. Once the source is added it will get a different order which makes it last in the list.